extensions/amp-ad-network-base/0.1/amp-ad-network-base.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpAdContext} from './amp-ad-context';
+import {dev} from '../../../src/log';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {sendXhrRequest} from './amp-ad-utils';
+
+const TAG = 'amp-ad-network-base';
+
+export class AmpAdNetworkBase extends AMP.BaseElement {
+
+  constructor(element) {
+    super(element);
+
+    /** @private {Object<./amp-ad-type-defs.ValidatorResultType, !./amp-ad-render.Renderer>} */
+    this.boundRenderers_ = {};
+
+    /** @private {?./amp-ad-render.Validator} */
+    this.boundValidator_ = null;
+
+    /** @private {!./amp-ad-type-defs.LayoutInfoDef} */
+    this.initialSize_ = {
+      // TODO(levitzky) handle non-numeric values.
+      width: element.getAttribute('width'),
```

Comment:
Attributes are not guaranteed inside the constructor, you have to wait for `buildCallback`.

Reply:
Done.